core/common/src/main/java/alluxio/metrics/OperationSystemGaugeSet.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.metrics;
+
+import com.codahale.metrics.CachedGauge;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricSet;
+import com.sun.management.OperatingSystemMXBean;
+import com.sun.management.UnixOperatingSystemMXBean;
+
+import java.lang.management.ManagementFactory;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A set of counters for the os metric.
+ */
+public class OperationSystemGaugeSet implements MetricSet {
+
+  private OperatingSystemMXBean mOsmxb = (OperatingSystemMXBean)
```

Comment:
Different JVMs have differend underlying com.sun.management classes, we need to do a try catch logics, if we can cast to a, we register the corresponding metrics, if not, we try to cast to b and register the corresponding metrics, if cannot cast to both, we don't register anything

Reply:
Check it out here:
https://github.com/Alluxio/alluxio/blob/db01aae966849e88d342a71609ab3d910215afeb/core/common/src/main/java/alluxio/conf/PropertyKey.java#L3015-L3023